### DISCUSSION TITLE
连接自动断开并提示：connection reset

### DISCUSSION THREAD

**Discussion by zhouxin3200** [apache/kyuubi#4096] [Q&A]
当我一段时间不提交任务，过段时间提交的时候，系统会提示如下：
Unknown HS2 problem when communicating with Thrift server.
Error: org.apache.thrift.transport.TTransportException: java.net.SocketException: Connection reset (state=08S01,code=0)
只有当我在运行！reconnect的时候才能重新连接服务，请问设置哪个参数可以避免这种自动断开。

**Comment by pan3793** [comment]
It was explained in [FAQ](https://github.com/apache/kyuubi/discussions/2481)

> Kyuubi Server will close idle connection (kyuubi.session.idle.timeout, default is 6 hours), if you are using JDBC Connection Pool like Hikari, the connection obtained from the pool may hold an expired session handle, please do health check before using to avoid such issue.

**Reply by zhouxin3200** [reply]
thank you very much , it will be disconnect about 10 minute，and there is no error in the background.

**Reply by pan3793** [reply]
We have received a similar case caused by network firewall policy, the network topology looks like `beeline ==(firewall/vpn)==> kyuubi`, and the network policy will drop the idle socket.
